components/com_fabrik/models/form.php

```diff
@@ -2186,8 +2186,10 @@ public function validate()
 			foreach ($elementModels as $elementModel)
 			{
 				// If the user can't view or edit the element, then don't validate it. Otherwise user sees failed validation but no indication of what failed
-				if (!$elementModel->canUse() && !$elementModel->canView())
-				{
+				// if (!$elementModel->canUse() && !$elementModel->canView())
+				// Bauer note: This is wrong. For example, if !canUse, but canView, this would not be skipped because both are not false! 	
+				// It should be either... if (!$elementModel->canUse() || !$elementModel->canView()) - or...
```

Comment:
I wasn't trying to "fix" any "specific case". This revelation came about while I was trying to figure out why code is being run when it really doesn't need to be run- and slowing down all ajax calls. I already explained what is wrong with the coding logic- which explains why that is happening. (Well that, and checking for if(!empty($this->errors)) rather than if(!$ok) as explained below) I'm pretty sure this is a big reason why most ajax calls usually takes much longer than expected.